bot/cogs/snakes.py

```diff
@@ -0,0 +1,1208 @@
+
+        # Send it!
+        await ctx.send(
+            f"A wild {content['name'].title()} appears!",
+            file=File(final_buffer, filename=content['name'].replace(" ", "") + ".png")
+        )
+
+    @command(name="snakes.fact()", aliases=["snakes.fact"])
+    async def snake_fact(self, ctx: Context):
+        """
+        Gets a snake-related fact
+
+        Written by Andrew and Prithaj.
+        Modified by lemon.
+        """
+
+        # Get a fact from the API.
+        response = await self.bot.http_session.get(self.facts_url, headers=self.headers)
+        question = await response.json()
+
+        # Build and send the embed.
+        embed = Embed(
+            title="Snake fact",
+            color=SNAKE_COLOR,
+            description=question
+        )
+        await ctx.channel.send(embed=embed)
+
+    @command(name="snakes()", aliases=["snakes"])
+    async def snake_help(self, ctx: Context):
+        """
+        This just invokes the help command on this cog.
+        """
+
+        log.debug(f"{ctx.author} requested info about the snakes cog")
+        return await ctx.invoke(self.bot.get_command("help"), "Snakes")
+
+    @command(name="snakes.snakify()", aliases=["snakes.snakify"])
+    async def snakify(self, ctx: Context, message: str = None):
+        """
+        How would I talk if I were a snake?
+        :param ctx: context
+        :param message: If this is passed, it will snakify the message.
+                        If not, it will snakify a random message from
+                        the users history.
+
+        Written by Momo and kel.
+        Modified by lemon.
+        """
+
+        def predicate(message):
+            """
+            Check if the message was sent by the author.
+            """
+            return message.author == ctx.message.author
+
+        def get_random_long_message(messages, retries=10):
+            """
+            Fetch a message that's at least 3 words long,
+            but only if it is possible to do so in retries
+            attempts. Else, just return whatever the last
+            message is.
+            """
+            long_message = random.choice(messages)
+            if len(long_message.split()) < 3 or retries <= 0:
+                return get_random_long_message(messages, retries - 1)
+            return long_message
+
+        with ctx.typing():
+            embed = Embed()
+            user = ctx.message.author
+
+            if not message:
+
+                # Get a random message from the users history
+                messages = []
+                async for message in ctx.channel.history(limit=500).filter(predicate):
+                    messages.append(message.content)
+
+                message = get_random_long_message(messages)
+
+            # Set the avatar
+            if user.avatar is not None:
+                avatar = f"https://cdn.discordapp.com/avatars/{user.id}/{user.avatar}"
+            else:
+                avatar = (
```

Comment:
Is there any reason we can't use the users default avatar (colored clyde image).

The way to get the link to these is through the `ctx.author.default_avatar_url` attribute.